### PR TITLE
Lottie enhancements

### PR DESCRIPTION
**Pull request by arinaturcu**
SUMMARY
---

- The Centered horizontal alignment of the Lottie block in the editor is fixed
- New background color setting
- Width control can now be set to `px` or `%`
- Maximum width in `%` is `100%`, so any value grater than that will fallback to `100%`
- New Aria label setting
- The `None` option from the animation trigger selector is now named `Autoplay`

TEST INSTRUCTIONS
---

It's important that the users that already use the Lottie block won't get any errors after updating and everything should work as before.

The main thing that may be bug-prone and should be tested is the Width range control.

### CODE REVIEW

**Comment by irinelenache** [comment]
@arinaturcu Good job 🚀 , tested and here are my findings:

- [ ] When the **animation trigger** is set to **scroll**, the background color won't work anymore https://vertis.d.pr/mYCGGI
- [ ] Found something related to the number of loops. If i set it to 3, without the reverse activated, the animation will loop 4 times. If i set it to 3, with reverse activated, the animation will loop 2 times https://vertis.d.pr/QXQkTk

Besides this, everything seems to work as expected 👍

File: src/blocks/blocks/lottie/block.json [hunk 1]
```diff
@@ -36,6 +36,19 @@
...
 			"type": "number"
+		},
+		"widthUnit": {
+			"type": "string"
+		},
+		"ariaLabel": {
+			"type": "string",
+			"default": ""
+		},
+		"bgColor": {
```
Comment (HardeepAsrani) [inline src/blocks/blocks/lottie/block.json:47]: For the sake of consistency, you can use the attribute name as `backgroundGradient` and `backgroundColor`

**Comment by HardeepAsrani** [comment]
@irinelenache Make sure to test it from old version to new when Background Color and Width Unit is selected.

**Comment by HardeepAsrani** [comment]
@arinaturcu Maybe you can use try to use this for the width unit? https://wordpress.github.io/gutenberg/?path=/story/components-experimental-unitcontrol--default - you can see if you can make it backward compatible.

**Comment by HardeepAsrani** [comment]
I suppose we can put the Aria Label to Advanced.

**Comment by arinaturcu** [comment]
@HardeepAsrani I used the UnitControl you mentioned and it is backward compatible so everything is fine. Can you review the changes?

File: src/blocks/blocks/lottie/block.json [hunk 2]
```diff
@@ -35,7 +35,17 @@
 			"default": false
 		},
 		"width": {
-			"type": "number"
+			"type": [ "string" , "number" ]
```
Comment (HardeepAsrani) [inline src/blocks/blocks/lottie/block.json:38]: @arinaturcu Didn't know you could do that. Does this work? Thanks, learned something new.

File: src/blocks/blocks/lottie/index.js [hunk 3]
```diff
@@ -32,3 +36,30 @@ registerBlockType( name, {
 	edit,
 	save
 });
+
+const lottieAdvancedControls = createHigherOrderComponent( BlockEdit => {
```
Comment (HardeepAsrani) [inline src/blocks/blocks/lottie/index.js:40]: @arinaturcu I don't think you need to use filters for that. You should be able to use `InspectorAdvancedControls` straight where you use InspectorControls.

File: src/blocks/blocks/lottie/block.json (hunk 2 above)
Comment (arinaturcu) [reply src/blocks/blocks/lottie/block.json:38]: Yes, it seems to work. I was a bit surprised too :smile:

File: src/blocks/blocks/lottie/index.js (hunk 3 above)
Comment (arinaturcu) [reply src/blocks/blocks/lottie/index.js]: Oh, that's right.

**Comment by arinaturcu** [comment]
@irinelenache Thank you for reviewing :rocket: 

There were some changes here :smile:
Can you test again?

File: src/blocks/blocks/lottie/block.json (hunk 2 above)
Comment (Soare-Robert-Daniel) [reply src/blocks/blocks/lottie/block.json:38]: :0, I can not believe we can use union types.

**Pull request merged**